aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -1075,131 +1083,108 @@ static bool _scaled_mm_is_fnuz() {
 
-    auto ceil_div = [](auto a, auto b) { return (a + b - 1) / b; };
-    auto num_k_blocks = ceil_div(dim_k_unpacked, BLOCK_SIZE_K);
-    auto padded_num_k_blocks = ceil_div(num_k_blocks, 4) * 4;
+    auto num_inner_blocks = ceil_div(inner_dim_unpacked, INNER_BLOCK_SIZE);
+    auto padded_num_inner_blocks = round_up(num_inner_blocks, 4);
 
     // TODO: We might want to enforce some structure on the shapes of the scale
     // tensors
 
     // Check expected sizes for block-wise scaling
-    auto expected_a_size =
-        BLOCK_SIZE_MN * ceil_div(dim_m, BLOCK_SIZE_MN) * padded_num_k_blocks;
-    auto expected_b_size =
-        BLOCK_SIZE_MN * ceil_div(dim_n, BLOCK_SIZE_MN) * padded_num_k_blocks;
-
-    TORCH_CHECK(scale_a.numel() == expected_a_size,
-                "For BlockWise scaling: Expected scale_a size to be ",
-                expected_a_size, " but got ", scale_a.numel());
-    TORCH_CHECK(scale_b.numel() == expected_b_size,
-                "For BlockWise scaling: Expected scale_b size to be ",
-                expected_b_size, " but got ", scale_b.numel());
+    auto expected_size = round_up(outer_dim, OUTER_BLOCK_SIZE) * padded_num_inner_blocks;
+
+    TORCH_CHECK(scale.numel() == expected_size,
+                "For BlockWise scaling: Expected scale_", operand_name, " size to be ",
+                expected_size, " but got ", scale.numel());
 
     TORCH_CHECK(
-        scale_a.is_contiguous() && scale_b.is_contiguous(),
-        "For BlockWise scaling: Both scale_a and scale_b must be contiguous");
+        scale.is_contiguous(),
+        "For BlockWise scaling: scale_", operand_name, " must be contiguous");
 
-    return ScalingType::BlockWise;
+    return INNER_BLOCK_SIZE == 16 ? ScalingType::BlockWise1x16 : ScalingType::BlockWise1x32;
   }
   // Both Per-Tensor and Row-wise scaling expect fp32 tensors
   TORCH_CHECK(
-      scale_a.scalar_type() == kFloat && scale_b.scalar_type() == kFloat,
-      "Both scale_a and scale_b must be float (fp32) tensors.");
+      scale.scalar_type() == kFloat,
+      "scale_", operand_name, " must be a float (fp32) tensor");
 
   // Check the singluar scale case for per-tensor scaling
-  if (scale_a.numel() == 1 && scale_b.numel() == 1) {
+  if (scale.numel() == 1) {
     return ScalingType::TensorWise;
   }
 
   // For non-TensorWise scaling, enforce 2D input tensors
   TORCH_CHECK(
-      scale_a.dim() == 2 && scale_b.dim() == 2,
+      scale.dim() == 2,
       "For non-TensorWise scaling, scale tensors must be 2-dimensional, "
-      "but got scale_a.dim()=",
-      scale_a.dim(),
-      " and scale_b.dim()=",
-      scale_b.dim());
+      "but got scale_", operand_name, ".dim()=", scale.dim());
 
   // Check for RowWise scaling
-  if (scale_a.size(0) == dim_m && scale_a.size(1) == 1 &&
-      scale_b.size(0) == 1 && scale_b.size(1) == dim_n) {
-#if (!defined(USE_ROCM) && !defined(_MSC_VER)) || \
-    (defined(USE_ROCM) && (defined(HIPBLASLT_VEC_EXT) || defined(HIPBLASLT_OUTER_VEC)))
```

Comment:
All of this is checked inside the lower-level CUDABlas.cpp function, I felt it redundant to have it here too.